torch_xla/csrc/ops/count_nonzero.cpp

```diff
@@ -0,0 +1,48 @@
+#include "torch_xla/csrc/ops/count_nonzero.h"
+
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/xla_lower_util.h"
+
+namespace torch_xla {
+namespace {
+
+xla::Shape NodeOutputShape(const torch::lazy::Value& input,
+                           std::vector<int64_t> dims) {
+  xla::Shape input_shape = GetXlaShape(input);
+  std::vector<int64_t> dimensions;
+  if (!dims.empty()) {
+    std::unordered_set<int64_t> dims_set(dims.begin(), dims.end());
+    for (int64_t i = 0; i < input_shape.rank(); i++) {
+      if (dims_set.find(i) != dims_set.end()) {
+        dimensions.push_back(i);
```

Comment:
nit, dims here should already be unique given the check in `XLANativeFunctions::count_nonzero`. If all you want is to sort it why not just `std::sort (dims.begin(), dims.end())`

are you also trying to handle the case that `dims` contains `value >= input_shape.rank()` ?  Is that a valid case for this op?

Reply:
right, it was meant to remove duplicate dimensions. After I added the check, this is not needed anymore.